src/inline-assembly.md

```diff
@@ -37,515 +50,878 @@ assert_eq!(x, 4 * 6);
-- SIMD vectors (structs defined with `#[repr(simd)]` and which implement `Copy`).
-This includes architecture-specific vector types defined in `std::arch` such as `__m128` (x86) or `int8x16_t` (ARM).
-
-Here is the list of currently supported register classes:
-
-| Architecture | Register class | Registers | LLVM constraint code |
-| ------------ | -------------- | --------- | -------------------- |
-| x86 | `reg` | `ax`, `bx`, `cx`, `dx`, `si`, `di`, `bp`, `r[8-15]` (x86-64 only) | `r` |
-| x86 | `reg_abcd` | `ax`, `bx`, `cx`, `dx` | `Q` |
-| x86-32 | `reg_byte` | `al`, `bl`, `cl`, `dl`, `ah`, `bh`, `ch`, `dh` | `q` |
-| x86-64 | `reg_byte`\* | `al`, `bl`, `cl`, `dl`, `sil`, `dil`, `bpl`, `r[8-15]b` | `q` |
-| x86 | `xmm_reg` | `xmm[0-7]` (x86) `xmm[0-15]` (x86-64) | `x` |
-| x86 | `ymm_reg` | `ymm[0-7]` (x86) `ymm[0-15]` (x86-64) | `x` |
-| x86 | `zmm_reg` | `zmm[0-7]` (x86) `zmm[0-31]` (x86-64) | `v` |
-| x86 | `kreg` | `k[1-7]` | `Yk` |
-| x86 | `kreg0` | `k0` | Only clobbers |
-| x86 | `x87_reg` | `st([0-7])` | Only clobbers |
-| x86 | `mmx_reg` | `mm[0-7]` | Only clobbers |
-| x86-64 | `tmm_reg` | `tmm[0-7]` | Only clobbers |
-| AArch64 | `reg` | `x[0-30]` | `r` |
-| AArch64 | `vreg` | `v[0-31]` | `w` |
-| AArch64 | `vreg_low16` | `v[0-15]` | `x` |
-| AArch64 | `preg` | `p[0-15]`, `ffr` | Only clobbers |
-| ARM (ARM/Thumb2) | `reg` | `r[0-12]`, `r14` | `r` |
-| ARM (Thumb1) | `reg` | `r[0-7]` | `r` |
-| ARM | `sreg` | `s[0-31]` | `t` |
-| ARM | `sreg_low16` | `s[0-15]` | `x` |
-| ARM | `dreg` | `d[0-31]` | `w` |
-| ARM | `dreg_low16` | `d[0-15]` | `t` |
-| ARM | `dreg_low8` | `d[0-8]` | `x` |
-| ARM | `qreg` | `q[0-15]` | `w` |
-| ARM | `qreg_low8` | `q[0-7]` | `t` |
-| ARM | `qreg_low4` | `q[0-3]` | `x` |
-| RISC-V | `reg` | `x1`, `x[5-7]`, `x[9-15]`, `x[16-31]` (non-RV32E) | `r` |
-| RISC-V | `freg` | `f[0-31]` | `f` |
-| RISC-V | `vreg` | `v[0-31]` | Only clobbers |
-| LoongArch | `reg` | `$r1`, `$r[4-20]`, `$r[23,30]` | `r` |
-| LoongArch | `freg` | `$f[0-31]` | `f` |
+operand := reg_operand / sym_operand / clobber_abi / options
+asm_inner := format_string *("," format_string) *("," operand) [","]
+asm := "asm!(" asm_inner ")"
+global_asm := "global_asm!(" asm_inner ")"
+
+non_format_char := ANY_CHAR // except "{" and "}"
+operand_specifier := ident / DEC_LITERAL
+expansion_specifier := *non_format_char
+format_specifier := "{" [operand_specifier] [":" *expansion_specifier]  "}"
+format_escape := "{{" / "}}"
+asm_string_piece := non_format_char / format_specifier / format_escape
+asm_string_content := [*asm_string_piece]
+```
+
+## Invocation [asm.invocation]
+
+r[asm.invocation.asm]
+The [`core::arch::asm!`] macro shall be expanded in an expression context only. The input tokens shall match the `asm_inner` production. The expansion is [`unsafe`][static.expr.safety] and has type `()`, unless the option `noreturn` is specified, in which case it has type `!`.
+
+```rust
+# #[cfg(target_arch = "x86_64")] 
+pub fn main(){
```

Comment:
```suggestion
pub fn main() {
```

It would be good to generally try to ensure that the examples are formatted in a way that is pretty close to how rustfmt works.